scripts/cc-cli/src/utils/tx.ts

```diff
@@ -74,3 +75,44 @@ export interface TxResult {
   status: TxStatus;
   info: string;
 }
+
+export async function getTxFee(
+  tx: SubmittableExtrinsic<"promise", ISubmittableResult>,
+  callerAddress: string
+): Promise<BN> {
+  const fee = await tx.paymentInfo(callerAddress);
+  return fee.partialFee.toBn();
+}
+
+export async function calculateBalanceAfterTx(
+  tx: SubmittableExtrinsic<"promise", ISubmittableResult>,
+  callerAddress: string,
+  amount: BN,
+  api: ApiPromise
+): Promise<BN> {
+  const balance = await getBalance(callerAddress, api);
+  const fee = await getTxFee(tx, callerAddress);
+  const availableBalance = balance.transferable;
+  const balanceAfterSending = availableBalance.sub(amount).sub(fee);
```

Comment:
Does BigNumber support negative values, or could this cause an overflow?

Reply:
BigNumber does support negative values!